command/taint.go

```diff
@@ -62,6 +64,34 @@ func (c *TaintCommand) Run(args []string) int {
 		return 1
 	}
 
+	// Load the config and check the core version requirements are satisfied
```

Comment:
I'm not sure, but I'd be curious to see if the `earlyconfig` loader (examples in `init.go`) is sufficient and perhaps faster for this purpose.  `loadSingleModule` for e.g. will only load the root module:

https://github.com/hashicorp/terraform/blob/6d86cd46f7394e4d655a18fc2e977f1c670f668e/command/meta_config.go#L95-L97

Again, this might not actually be faster or better, but it might be worth looking at.

Reply:
Ah, I hadn't thought of that! Unfortunately I think we need to load and check the entire tree's version constraints. In particular, many public registry modules have fairly strict version requirements that we need to honour.